### PR TITLE
JESD health checks

### LINKED ISSUE #467: Add additional constrains to the JESD health check method
## Describe the problem
We should add the following additional constrains to the Jesd health method:
```
JesdTx.DataValid (==1 per active channel)
JesdTx.StatusValidCnt (== 0 per active channel)
JesdRx.DataValid (==1 per active channel)
JesdRx.StatusValidCnt (== 0 per active channel)
JesdRx.DisparityErr (==0 per active channel)
JesdRx.NotInTableErr (==0 per active channel)
DAC.LinkErrCnt[3:0] == 0
DAC.ReadFifoEmpty[3:0] == 0
DAC.ReadFifoUnderflow[3:0] == 0
DAC.ReadFifoFull[3:0] == 0
DAC.ReadFifoOverflow[3:0] == 0
DAC.DispErr[3:0] == 0
DAC.NotitableErr[3:0] == 0
DAC.CodeSyncErr[3:0] == 0
DAC.FirstDataMatchErr[3:0] == 0
DAC.ElasticBuffOverflow[3:0] == 0
DAC.LinkConfigErr[3:0] == 0
DAC.FrameAlignErr[3:0] == 0
DAC.MultiFrameAlignErr[3:0] == 0
```

If these checks fails, subsequence data-taking operations should not be performed as they will produce incorrect data. Instead, the system should be configured again by calling the `set_defaults` command, which will trigger a call to `AppTop.Init()` to re-establish the JESD links.

### PR DESCRIPTION
**Pull request by eyyoung24**
## Issue
This PR resolves #467. 

## Description

- This implements the new JESD checks.  These features are only usable for pysmurf core code version >4.1.0 (not yet released) and rogue ZIP file versions >0.3.0.
- This adds a `set_check_jesd` to `_set_defaults_cmd`
- changes defaults subbands in `find_freq` to None. This does not change the interface because if None, then it uses the same subbands as before (13, 115)
- changes the default noise bins in `take_noise_psd` to None. If None, loads the same defaults as before, thus no interface change. 

### Useful note
Stolen from this comment (https://github.com/slaclab/pysmurf/issues/467#issuecomment-664688186): 
- `CheckJesd`: it is a command used to trigger a call to the new `AppTop.JesdHealth` method. The command will check if the ZIP file's `AppTop` device contains the new `JesdHealth` method, call it if it exists, and catch the returned value.
- `JesdStatus`: this will contain the status of the `CheckJesd` command. It is an `enum` register with these states: 
  - `Unlocked`: The `AppTop.JesdHealth` said the JESD were not locked,
  - `Locked`: The `AppTop.JesdHealth` said the JESD were locked,
  - `Checking`: The `AppTop.JesdHealth` command is in progress, and 
  - `Not found`: The ZIP file does not contains the `AppTop.JesdHealth`  method.

## Does this PR break any interface?
- [ ] Yes
- [x] No

### CODE REVIEW

File: python/pysmurf/client/command/smurf_command.py [hunk 1]
```diff
@@ -2729,6 +2729,190 @@ def get_jesd_tx_status_valid_cnt(self, bay, num, **kwargs):
...
+        # Extra registers were added to allow confirmation of JESD
+        # lock for pysmurf versions >=4.1.0.
+        # https://github.com/slaclab/pysmurf/issues/467 for more
+        # details.
+        if version.parse(pysmurf_version) >= version.parse('4.1.0'):
+            status =  self._caget(
+                self.smurf_application + self._jesd_status_reg,
+                as_string=as_string, **kwargs)
+
+            if status == 'Not found':
```
Comment (jesusvasquez333) [inline python/pysmurf/client/command/smurf_command.py:2902]: If you call this method with `S.get_jesd_status(as_string=False)` this statement will failed, as status will be a number.

File: python/pysmurf/client/command/smurf_command.py [hunk 2]
```diff
@@ -2729,6 +2729,190 @@ def get_jesd_tx_status_valid_cnt(self, bay, num, **kwargs):
...
+                'The `SmurfApplication:CheckJesd` and '
+                ' `SmurfApplication:JesdStatus` registers are not'
+                ' implemented for pysmurf core code versions <4.1.0'
+                f' (current version is {pysmurf_version}).',
+                self.LOG_ERROR)
+            return None
+
+    _jesd_status_reg = "JesdStatus"
+
+    def get_jesd_status(self, as_string=True, **kwargs):
```
Comment (jesusvasquez333) [inline python/pysmurf/client/command/smurf_command.py:2850]: You need to remove the `as_string` argument from this method, due to the comments I added above.

**Comment by jesusvasquez333** [comment]
@swh76 I’m testing with this branch of pysmurf and ZIP `v0.2.0`: Right after starting the server + pysmurf client I see:
```python
In [3]: epics.caget(f'{epics_prefix}:AMCc:SmurfApplication:JesdStatus', as_string=True)
Out[3]: 'Not found'
```
I see the same on the GUI.

Then, after I do `S = pysmurf.client.SmurfControl(epics_root=epics_prefix,cfg_file=config_file,setup=True,make_logfile=False,shelf_manager="shm-smrf-sp01")` I see:
```python
In [5]: epics.caget(f'{epics_prefix}:AMCc:SmurfApplication:JesdStatus', as_string=True)
Out[5]: 'Not found'
```

Same on the GUI.

Then I tried you new command:
```python
In [6]: s = S.get_jesd_status()
[ 2020-08-04 16:48:24 ]  The `JesdHealth` method is not present in the Rogue ZIP file.
In [7]: print(s)
Not found
```

This one as well:
```python
In [8]: s = S.set_check_jesd()
[ 2020-08-04 16:49:17 ]  The `JesdHealth` method is not present in the Rogue ZIP file.
[ 2020-08-04 16:49:17 ]  The `JesdHealth` method is not present in the Rogue ZIP file.
In [9]: print(s)
Not found
```

So, it looks all good to me.

**Comment by jesusvasquez333** [comment]
@swh76 Now I'm trying with ZIP `v0.3.0`. Right after I start the system I see:
```python
In [1]: import epics

In [2]: epics.caget(f'{epics_prefix}:AMCc:SmurfApplication:JesdStatus', as_string=True)
Out[2]: 'Unlocked'
```

I see the same on the GUI.

Then, after I run `S = pysmurf.client.SmurfControl(epics_root=epics_prefix,cfg_file=config_file,setup=True,make_logfile=False,shelf_manager="shm-smrf-sp01")` I see:
```python
In [4]: epics.caget(f'{epics_prefix}:AMCc:SmurfApplication:JesdStatus', as_string=True)
Out[4]: 'Locked'
```
and same on the GUI.

Now I tried your new command:
```python
In [5]: s = S.get_jesd_status()

In [6]: print(s)
Locked
```

and:
```python
In [7]: s = S.set_check_jesd()
[ 2020-08-04 17:12:22 ]  JESD health check finished after 1 seconds. The final status was Locked.

In [8]: print(s)
Locked
```

All looks good here as well.

File: python/pysmurf/client/command/smurf_command.py (hunk 2 above)
Comment (eyyoung24) [reply python/pysmurf/client/command/smurf_command.py]: removed `as_string` opt args

**Pull request merged**